gadget/device_linux.go

```diff
@@ -54,11 +54,23 @@ func FindDeviceForStructure(ps *PositionedStructure) (string, error) {
 	var found string
 	var match string
 	for _, candidate := range candidates {
-		if !osutil.FileExists(candidate) {
-			continue
+		st, err := os.Lstat(candidate)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			return "", fmt.Errorf("cannot stat device: %v", err)
+		}
+		if (st.Mode() & os.ModeSymlink) == 0 {
+			// /dev/disk/by-label/* and /dev/disk/by-partlabel/* are
+			// expected to be symlink
```

Comment:
Is this an implementation detail or an advertised feature of udev?

Reply:
It's part of persistent device naming. Though I don't recall seeing any official docs on this aide from bits of information scattered around different wikis, one is here https://wiki.archlinux.org/index.php/Persistent_block_device_naming